### PR TITLE
[JENKINS-50296] Run ThreadPoolForRemoting as SYSTEM

### PR DESCRIPTION
**Pull request by agentgonzo**
See [JENKINS-50296](https://issues.jenkins-ci.org/browse/JENKINS-50296).

Changed the `threadPoolForRemoting` to run in the context of the `SYSTEM` user (at present, it runs as `ANONYMOUS`) to keep the code consistent with executors - see #2792 

@reviewbybees

### CODE REVIEW

**Review by jglick — APPROVED** [review]
Makes sense I think. Note that there is a similarly-named yet distinct thread pool in `FilePath`, used only for`actAsync` on a non-remote file. (`FilePath` cannot refer to `Computer`’s version since that could blow up if accessed from the agent side.)

**Review by oleg-nenashev — COMMENTED** [review]
threadPoolForRemoting is being heavily misused in plugins: https://issues.jenkins-ci.org/browse/JENKINS-27514 and https://issues.jenkins-ci.org/browse/JENKINS-47258 . 

I think that this PR is generally fine, but IMHO it will need serious soak testing before backporting.

Some autotests demonstrating the fix are also required in this case IMHO

**Comment by oleg-nenashev** [comment]
@agentgonzo BTW JENKINS-50296 is submitted as improvement. So maybe my comment about backporting is not actual

File: .mvn/jvm.config [hunk 1]
```diff
@@ -1 +0,0 @@
--Xmx800m
```
Comment (oleg-nenashev) [inline .mvn/jvm.config]: hm?

**Review by oleg-nenashev — APPROVED** [review]

File: .mvn/jvm.config (hunk 1 above)
Comment (agentgonzo) [reply .mvn/jvm.config]: yeah, that shouldn't be there!

Comment (agentgonzo) [reply .mvn/jvm.config]: fixed now.

**Comment by daniel-beck** [comment]
Won't this potentially elevate bugs in plugins to security vulnerabilities?

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
> Won't this potentially elevate bugs in plugins to security vulnerabilities?

Always possible, but strikes me as unlikely. Would require there to have been an existing call which is sensitive to access control which either passed as `ANONYMOUS` (in which case we are not changing anything), or failed (in which case presumably the plugin author would have noticed already).

**Comment by oleg-nenashev** [comment]
On hold until @daniel-beck confirms that he is fine from the Security PoV

File: core/src/test/java/hudson/model/ComputerTest.java [hunk 2]
```diff
@@ -28,4 +37,11 @@ public void testRelocate() throws Exception {
             dir.deleteRecursive();
         }
     }
+
+    @Issue("JENKINS-42556")
```
[the remark this answers is not in the document]
Comment (agentgonzo) [reply core/src/test/java/hudson/model/ComputerTest.java]: fixed now.

**Comment by daniel-beck** [comment]
Discussed with @jglick @agentgonzo @jtnord; we don't expect security problems here. Use of `readResolve` permission checks is limited to XStream in threads associated with a user authentication for something like `POST config.xml`.

**Review by jtnord — APPROVED** [review]

**Comment by agentgonzo** [comment]
@daniel-beck - is this OK to merge now? I don't have the permissions to do so.

**Comment by agentgonzo** [comment]
@jglick @daniel-beck - could someone merge this please?

**Pull request merged**